### ISSUE TITLE
Inconsistent `[vt-start vt-end]` should not be indexed and get aborted 

### ISSUE THREAD

**Issue by FiV0** [xtdb/crux#2494]
Currently one can submit certain valid-time ranges that don't make sense. These transactions should not get indexed.
```clj
(t/deftest test-inconsistent-valid-time-range 
  (xt/submit-tx tu/*node* '[[:put :xt_docs {:xt/id :mark} {:for-valid-time [:in nil #inst "2011"]}]])
  (t/is (= []
           (xt/q tu/*node*
                 '{:find [id vt-start vt-end], :where [(match :xt_docs {:xt/id id
                                                                        :xt/valid-from vt-start
                                                                        :xt/valid-to vt-end}
                                                              {:for-valid-time [:in nil #inst "2040"]})]}))))
```

**Comment by FiV0** [comment]
The test from above actually no longer surfaced the issue as it was also fixed with #2501. Using the `xt/txs` table instead.

**Comment by FiV0** [comment]
Fixed in #2505
